lightning/src/chain/keysinterface.rs

```diff
@@ -489,18 +514,10 @@ pub trait KeysInterface {
 	/// This method is slowly being phased out -- it will only be called when reading objects
 	/// written by LDK versions prior to 0.0.113.
 	fn read_chan_signer(&self, reader: &[u8]) -> Result<Self::Signer, DecodeError>;
+}
 
-	/// Sign an invoice.
-	/// By parameterizing by the raw invoice bytes instead of the hash, we allow implementors of
-	/// this trait to parse the invoice and make sure they're signing what they expect, rather than
-	/// blindly signing the hash.
-	/// The hrp is ascii bytes, while the invoice data is base32.
-	///
-	/// The secret key used to sign the invoice is dependent on the [`Recipient`].
-	///
-	/// Errors if the `Recipient` variant is not supported by the implementation.
-	fn sign_invoice(&self, hrp_bytes: &[u8], invoice_data: &[u5], receipient: Recipient) -> Result<RecoverableSignature, ()>;
-
+/// A trait to describe an object which can get user secrets and key material.
+pub trait KeysInterface: EntropySource + NodeSigner + SignerProvider {
```

Comment:
Yeah, the idea is to stop having KeysInterface implement the other traits. This is just the first step, because doing all the refactoring in one PR is a bit unoverviewable.